hivemind/utils/mpfuture.py

```diff
@@ -161,4 +161,5 @@ def __await__(self):
 
     def __del__(self):
         self._shutdown_trigger.set_result(True)
-        self.connection.close()
+        if hasattr(self, 'connection'):
+            self.connection.close()
```

Comment:
The issue arises when connection destructor is called in parallel to its creation. While this does not *normally* happen, it may sometimes occur in DHT and averager when handling unresponsive peers